keps/693-multikueue/README.md

```diff
@@ -0,0 +1,234 @@
+
+## Summary
+Introduce an new AdmissionCheck (called MultiKueue) with dedicated API
+and controller that will provide multi-cluster capabilities to Kueue.
+
+## Motivation
+Many of Kueue's users are running multiple clusters and would like to
+have a way to easily distribute batch jobs across them to keep all of
+them utilized. Without a global distribution point, some clusters may
+get less jobs they are able to process, while the others get more, leading
+to underutilization and higher costs. 
+
+### Goals
+* Allow Kueue distribute batch jobs across multiple clusters,
+while maintaining the specified quota limits.
+* Provide users with a single entry point through which the jobs
+can be submitted and monitored, just like they were running in
+a single cluster.
+* Be compatible with all Kueue's features (priorities, borrowing, preemptions, etc)
+and most of integrations.
+* Allow to upgrade single cluster Kueue deployments to multicluster without
+much hassle.
+
+### Non-Goals
+* Solve storage problem. It is assumed that the distributed jobs are
+either location-flexible (for a subset of clusters) or are copying the 
+data as a part of the startup process.
+* Synchronize configuration across the clusters. It is expected that the 
+user will create the appropriate objects, roles and permissions
+in the clusters (manually, using gitops or some 3rd-party tooling).
+* Set up authentication between clusters.
+
+## Proposal
+
+Introduce MultiKueue AdmissionCheck, controller and configuration API. 
+
+Establish the need for a designated management cluster.
+
+![Architecture](arch.png "Architecture")
+
+For each workload coming to a ClusterQueue (with the MultiKueue AdmissonCheck enabled)
+in the management cluster, and getting past the preadmission phase in the 
+two-phase admission process (meaning that the global quota is ok), 
+MultiKueue controller will create it in the defined worker clusters and wait 
+until some Kueue running there admits the workload. 
+Then it will remove the workloads from the remaining clusters and allow the
+single instance of the job to proceed. The workload will be also admitted in 
+the management cluster.
+
+There will be no job controllers running in the management clusters or they will be
+disabled for the workloads coming to MultiKueue-enabled cluster queues via annotation
+or some other, yet to be decided, mechanism. By disabling we mean that the controller
+will do no action on the selected objects, no pods (or other objects created and 
+allowing other controllets to update their status as they see fit.
+
+There will be just CRD/job definitions deployed. MultiKueue controller will copy the status
+of the job from the worker clusters, so that it will appear that the job
+is running inside the management clusters. However, as there is no job controller,
+no pods will be created in the management cluster. No controller will also overwrite
+the status that will be copied by the MultiKueue controller. 
+
```

Comment:
```suggestion
If the job is running in a remote cluster, (workloads admission check status is set to Retry), the job will remain suspended and the status of the remote job will be copied at the end of the remote execution.

The runtime status of a remote job can be checked on the remote cluster, keeping open the option of
extending the visibility API or kueue CLI to ease this operation.
```

Reply:
The status of the job needs to be copied along the execution, not only at the end.